keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -133,130 +132,114 @@ class ClusterHandler(
+        "loadBalancers" to dependencies.loadBalancerNames,
+        "targetGroups" to dependencies.targetGroups,
+        "account" to location.accountName
+      )
+    }
+      .also { job ->
+        current?.also { ancestor ->
           job["source"] = mapOf(
-            "account" to location.accountName,
-            "region" to location.region,
-            "asgName" to ancestorServerGroup.asg.autoScalingGroupName
+            "account" to ancestor.location.accountName,
+            "region" to ancestor.location.region,
+            "asgName" to ancestor.moniker.serverGroup
           )
           job["copySourceCustomBlockDeviceMappings"] = true
         }
-    }
-
-  private suspend fun ServerGroup.resizeServerGroupJob(serviceAccount: String): Map<String, Any?> =
-    cloudDriverService.getAncestorServerGroup(this, serviceAccount)
-      ?.let { currentServerGroup ->
-        mapOf(
-          "type" to "resizeServerGroup",
-          "capacity" to mapOf(
-            "min" to capacity.min,
-            "max" to capacity.max,
-            "desired" to capacity.desired
-          ),
-          "cloudProvider" to currentServerGroup.cloudProvider,
-          "credentials" to location.accountName,
-          "moniker" to mapOf(
-            "app" to currentServerGroup.moniker.app,
-            "stack" to currentServerGroup.moniker.stack,
-            "detail" to currentServerGroup.moniker.detail,
-            "cluster" to currentServerGroup.moniker.cluster,
-            "sequence" to currentServerGroup.moniker.sequence
-          ),
-          "region" to currentServerGroup.region,
-          "serverGroupName" to currentServerGroup.asg.autoScalingGroupName
-        )
       }
-      ?: throw ResourceConflict("Could not find current server group for cluster ${moniker.name} in ${location.accountName} / ${location.region}")
+
+  private fun ResourceDiff<ServerGroup>.resizeServerGroupJob(): Map<String, Any?> {
+    val current = requireNotNull(current) {
+      "Current server group must not be null when generating a resize job"
+    }
+    return mapOf(
+      "type" to "resizeServerGroup",
+      "capacity" to mapOf(
+        "min" to desired.capacity.min,
+        "max" to desired.capacity.max,
+        "desired" to desired.capacity.desired
+      ),
+      "cloudProvider" to CLOUD_PROVIDER,
+      "credentials" to desired.location.accountName,
+      "moniker" to mapOf(
+        "app" to current.moniker.app,
+        "stack" to current.moniker.stack,
+        "detail" to current.moniker.detail,
+        "cluster" to current.moniker.name,
+        "sequence" to current.moniker.sequence
+      ),
+      "region" to current.location.region,
+      "serverGroupName" to current.name
+    )
+  }
 
   override suspend fun delete(resource: Resource<ClusterSpec>) {
     TODO("not implemented")
   }
 
-  private suspend fun CloudDriverService.getAncestorServerGroup(spec: ServerGroup, serviceAccount: String): ActiveServerGroup? =
```

Comment:
No longer need to look this up as we have it already from having determined current cluster state.